### PR TITLE
Feature Flags

### PR DESCRIPTION
**Pull request by shagedorn**
First step towards resolving https://github.com/iosphere/ISHPermissionKit/issues/74.

Missing pieces:

* Create repo for sample app, which should use permission kit as submodule
* Fix test target
* Add CocoaPods subspecs

I've created `develop-feature-flags` to avoid merging this into `develop` before the obvious things are fixed.

### CODE REVIEW

File: ISHPermissionKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,56 +7,30 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		1B06A05A1D70949A00C7F8ED /* ISHPermissionKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = DC6A46B119D712AC00B11E88 /* ISHPermissionKit.framework */; };
-		1B06A05B1D70949A00C7F8ED /* ISHPermissionKit.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = DC6A46B119D712AC00B11E88 /* ISHPermissionKit.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
 		1B09324A1D7069C70095A74A /* ISHPermissionRequestContacts.m in Sources */ = {isa = PBXBuildFile; fileRef = 1B0932481D7069C70095A74A /* ISHPermissionRequestContacts.m */; };
 		1B09324C1D7069D90095A74A /* ISHPermissionRequestContacts.h in Headers */ = {isa = PBXBuildFile; fileRef = 1B09324B1D7069D90095A74A /* ISHPermissionRequestContacts.h */; };
 		1B09324D1D7069D90095A74A /* ISHPermissionRequestContacts.h in Headers */ = {isa = PBXBuildFile; fileRef = 1B09324B1D7069D90095A74A /* ISHPermissionRequestContacts.h */; };
-		1B09324E1D7069D90095A74A /* ISHPermissionRequestContacts.h in Headers */ = {isa = PBXBuildFile; fileRef = 1B09324B1D7069D90095A74A /* ISHPermissionRequestContacts.h */; };
```
Comment (shagedorn) [inline ISHPermissionKit.xcodeproj/project.pbxproj]: These files have not been removed. I assume these deletions stem from removing the test app.

File: ISHPermissionKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -879,31 +503,13 @@
 			projectRoot = "";
 			targets = (
 				DCC1BFA4195B174200362559 /* ISHPermissionKit */,
-				DC6A468819D712AC00B11E88 /* ISHPermissionKit+HealthKit */,
 				DCAEF7C9195C60070074EB5B /* ISHPermissionKitLib */,
-				DC6A465E19D7125300B11E88 /* ISHPermissionKitLib+HealthKit */,
-				DCC1BFC4195B176E00362559 /* PermissionsTestApp */,
 				DCC1BFAF195B174200362559 /* ISHPermissionKitTests */,
-				DC7790B219617CD7007CB2BC /* ISHPermissionKitLibTests */,
```
Comment (shagedorn) [inline ISHPermissionKit.xcodeproj/project.pbxproj]: I don't think we should keep two test targets that run the exact same tests...

File: README.md [hunk 3]
```diff
@@ -64,102 +70,122 @@ Please file an issue for missing permissions.
...
+*ISHPermissionKit* uses system frameworks to accomplish its tasks. Most of
+them will be linked automatically unless you have disabled "Enable Modules"
+(`CLANG_ENABLE_MODULES`) and "Link Frameworks Automatically" 
+(`CLANG_MODULES_AUTOLINK`) in your app target's build settings.
 
-ISHPermissionKit can also be installed as a framework through CocoaPods:
+Unfortunately, some framework are not weakly linked automatically which
+will cause your app to crash at launch on older systems that don't support
+the respective framework. These frameworks must be explicitly linked in
+your app, and set to "Optional". Feel free to duplicate rdar://28008958.
```
Comment (felixLam) [inline README.md]: We should maybe publish the radar on open radar to make it easier for others to dupe it: https://openradar.appspot.com/search?query=28008958

File: README.md [hunk 4]
```diff
@@ -64,102 +70,122 @@ Please file an issue for missing permissions.
...
 Add this Xcode project as a subproject of your app. Then link your app target 
-against the static library (`ISHPermissionKitLib.a`, or 
-`ISHPermissionKitLib+HealthKit.a` if you require HealthKit support).
-You will also need to add the static library as a target dependency. 
-Both settings can be found in your app target's *Build Phases*.
+against the static library (`ISHPermissionKitLib.a`). You will also need to add
+the static library as a target dependency. Both settings can be found in your 
+app target's *Build Phases*.
+
+You must [provide a build configuration](#providing-a-build-configuration) manually.
```
Comment (felixLam) [inline README.md]: Maybe make this entire paragraph bold?

File: README.md [hunk 5]
```diff
@@ -64,102 +70,122 @@ Please file an issue for missing permissions.
...
 
 Add this Xcode project as a subproject of your app. Then add the framework
-(`ISHPermissionKit.framework`, or `ISHPermissionKit+HealthKit.framework` if you 
-require HealthKit support) to the app's embedded binaries (on the *General*
+(`ISHPermissionKit.framework`) to the app's embedded binaries (on the *General*
 tab of your app target's settings). On the *Build Phases* tab, verify that the
 framework has also been added to the *Target Dependencies* and *Link Binary with
 Libraries* phases, and that a new *Embed Frameworks* phase has been created.
 
+You must [provide a build configuration](#providing-a-build-configuration) manually.
```
Comment (felixLam) [inline README.md]: Maybe make this entire paragraph bold?

File: README.md [hunk 6]
```diff
@@ -64,102 +70,122 @@ Please file an issue for missing permissions.
...
 
+### Providing a Build Configuration
+
+When building the static or dynamic library, *ISHPermissionKit* will look for an
+`.xcconfig` file in the same directory as *ISHPermissionKit*'s root directory 
+(not *within* the root directory). This file allows you to set preprocessor flags
+that will be used when compiling the framework.
+
+We strongly recommend to start with a copy of the template config provided in this
+repository, `ISHPermissionKitAppConfiguration.xcconfig`. It includes a list of
```
Comment (felixLam) [inline README.md]: Maybe add a link?

File: README.md (hunk 3 above)
Comment (shagedorn) [inline README.md]: Done

**Pull request merged**